src/firefly/js/tables/reducer/TableDataReduer.js

```diff
@@ -18,15 +20,18 @@ export function dataReducer(state={data:{}}, action={}) {
                 return update(root, { [tbl_id] : {selectInfo: {$set: selectInfo}}});
             } else return root;
 
+        case (Cntlr.TABLE_NEW_LOADED)  :
+            if (get(root, [tbl_id, 'tableMeta', 'Loading-Status'], 'COMPLETED') !== 'COMPLETED') {
+                return update(root, { [tbl_id] : {tableMeta: {['Loading-Status']:  {$set: 'COMPLETED'}}}});
```

Comment:
Regarding 'update', I agree the syntax is cryptic.  But, it's also a good idea to use already written code instead of writing it ourselves.  Why don't we wrap lodash 'set' with 'update' to create update funtions with lodash 'set' syntax..  ie.. using the above example...
        udSet(root, [tbl_id, 'tableMeta', 'Loading-Status'], 'COMPLETED');
   other use cases of update..
        udMerge((root, [tbl_id, 'tableData', 'data', rowIdx], newRowOfData);
        udSet((root, 'data.currrentTable.error', errorMsg);
            in this case, path is a string.